src/pieces/knight.js

```diff
@@ -1,16 +1,44 @@
+import Pawn from './pawn';
 import Piece from './piece';
 
+
 class Knight extends Piece {
   constructor(row, column, side) {
     super(row, column, side);
     this.name = 'knight';
     this.display = `<i class="fas fa-chess-knight ${side}"></i>`; //fontawesome knight
+  
   }
-  findLegalMoves() {
+  
+  findLegalMoves(board) {
     const possibleMoves = [];
 
+    const moves=[[-1,-2],[-2,-1],[-2,1],[-1,2],[1,-2],[1,2],[2,1],[2,-1]];
+    let moves_elements=[];
+
+    for (let i=0;i<moves.length;i++){
+        moves_elements = moves[i];
+        const ROW = moves_elements[0];
```

Comment:
I tak z innej beczki to ja bym to nazwał `rowOffset` bo ta zmienna nie reprezentuje nowego wiersza tylko bardziej o ile wierszów mam się przesunąć